### PR TITLE
changefeedccl: add production monitoring metrics

### PR DESCRIPTION
**Pull request by danhhz**
The following are added and registered via a ccl hook:

    changefeed.emitted_messages
    changefeed.emitted_bytes
    changefeed.flush_nanos
    changefeed.min_high_water

Release note (enterprise change): CHANGEFEEDS now export metrics for
production monitoring

### CODE REVIEW

**Comment by danhhz** [comment]
Paging @mrtracy, got a minute to look at this?

**Comment by mrtracy** [comment]
This looks good, the testing is good. The one thing I'll say is that the CCL magic was a bit hard to follow, resulting in lines like this:

`metrics := p.ExecCfg().JobRegistry.MetricsStruct().Changefeed.(*Metrics)`.

Which may seem overly complex, but which is perhaps the only viable solution. Some additional comments about that design may have helped me follow it; or maybe not, it didn't take that long to figure it out.

LGTM

**Review by danhhz — COMMENTED** [review]
It's not strictly ccl's fault. If we didn't have ccl and I wanted 1) a separate package for each job impl 2) to store the metrics for each job in that package and 3) needed to get access to the shared metrics from the job impl, I'd have the same problem. More of a jobs issues than a ccl issue, tbh. The most easily changed of these is (2) and I'm not opposed to moving the changefeeds metrics struct into the jobs package if you feel strongly. (Though as you point out, you did figured it out fairly quickly).

Anyway, threw in some comments but dunno how helpful they are. Also resolved a bunch of merge conflicts

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/28162#-:-LK2N56H1SKCwxaT2vTk:b-ttp8lu)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Comment by danhhz** [comment]
Thanks for the review!

bors r=mrtracy
